### ISSUE TITLE
Caffe2 installation error with protoc

### ISSUE THREAD

**Issue by ptzheng** [facebookarchive/caffe2#2417]
I followed the build from source instruction and encountered the following problem. How can I solve it?

### System information

* Operating system:Linux Ubuntu 14.04
* Compiler version:4.8
* CMake version:3.11.0-rc4
* CMake arguments:
* Relevant libraries/versions (e.g. CUDA):CUDA8.0+CUDNN7.0

******** Summary ********
[ 17%] Running C++ protocol buffer compiler on /home/ptzheng/caffe2/build/third_party/onnx/onnx/onnx_onnx_c2.proto 
Scanning dependencies of target onnx_proto 
[ 17%] Building CXX object third_party/onnx/CMakeFiles/onnx_proto.dir/onnx/onnx_onnx_c2.pb.cc.o 
In file included from /home/ptzheng/caffe2/build/third_party/onnx/onnx/onnx_onnx_c2.pb.cc:5:0: 
/home/ptzheng/caffe2/build/third_party/onnx/onnx/onnx_onnx_c2.pb.h:12:2: error: #error This file was generated by a newer version of protoc which is 
 #error This file was generated by a newer version of protoc which is 
  ^ 
/home/ptzheng/caffe2/build/third_party/onnx/onnx/onnx_onnx_c2.pb.h:13:2: error: #error incompatible with your Protocol Buffer headers. Please update 
 #error incompatible with your Protocol Buffer headers.  Please update 
  ^ 
/home/ptzheng/caffe2/build/third_party/onnx/onnx/onnx_onnx_c2.pb.h:14:2: error: #error your headers. 
 #error your headers. 
  ^ 
In file included from /home/ptzheng/caffe2/build/third_party/onnx/onnx/onnx_onnx_c2.pb.cc:5:0: 
/home/ptzheng/caffe2/build/third_party/onnx/onnx/onnx_onnx_c2.pb.h:23:35: fatal error: google/protobuf/arena.h: No such file or directory 
 #include <google/protobuf/arena.h> 
                                   ^ 
compilation terminated. 
make[2]: *** [third_party/onnx/CMakeFiles/onnx_proto.dir/onnx/onnx_onnx_c2.pb.cc.o] Error 1 
make[1]: *** [third_party/onnx/CMakeFiles/onnx_proto.dir/all] Error 2 
make: *** [all] Error 2

**Comment by zhangw864680355** [comment]
@ptzheng  maybe it is recaused by the protoc version .checkout the  protoc version: protoc --version
if it is 2.6.1 ,you need to update the new version,such as 3.5.1 ,by downloading the source code and install  instead of command. At the same times ,you had better to remove python-protobuf if you install it.

**Comment by pjh5** [comment]
Can you read https://caffe2.ai/docs/faq.html#how-do-i-fix-error-messages-that-are-protobuf-related and https://caffe2.ai/docs/faq.html#how-can-i-find-what-dependencies-my-caffe2-library-or-other-library-has and find your protobuf version mismatch?

**Comment by ptzheng** [comment]
I have read it before. But after I updated my protocol buffer, the new problem happened. The problem is that This file was generated by a newer version of protoc which is  incompatible with your Protocol Buffer headers...... 
I also tried another method. Firstly, I removed all the things related to protobuf by command as follows.
`sudo apt-get remove libprotobuf-dev protobuf-compiler
`sudo apt-get remove libprotobuf-lite8 libprotoc8
`sudo apt-get remove python-protobuf
 sudo pip uninstall protobuf 
 conda uninstall protobuf`
Then I installed it by conda `conda install -y -c conda-forge protobuf`. I used `protoc --version` and the result is `libprotobuf 3.5.1`  Howerver, when I make install again, the new problem is that it can't find my protobuf.
 I am a new for ubuntu. I don't kown the relationship betwenn protoc and protocol buffer headers. Can you please give me some adavice about matching the version between protoc and protoc buffer?

**Comment by zhangw864680355** [comment]
@ptzheng can you show me the error? find the path of protoc file  by "sudo find / -name protoc " and choose to use the protoc file in the source code to replace them,such path as /usr/local. 

wget https://github.com/google/protobuf/releases/protobuf-python-3.5.1.tar.gz
tar zxvf protobuf-python-3.5.1.tar.gz
cd protobuf-python-3.5.1.tar.gz
./configure
make
sudo make install

sudo find / -name protoc
cp xx to xx

if you can not compile and fail to translator ,continue to do

cd ./python
python setup.py build
python setup.py install

**Comment by ptzheng** [comment]
Thanks for your help, I have solved the problem above and compiled caffe2 successfully. I installed protobuf-3.3.0.
But when I try to test the caffe2 installation`cd ~ && python -c 'from caffe2.python import core' 2>/dev/null && echo "Success" || echo "Failure", it echoed Failure. I use `python -c 'from caffe2.python import core` to find the reason,which is "ImportError: No module named caffe2.python". 
My python information: Python 2.7.14 |Anaconda, Inc.| (default, Dec  7 2017, 17:05:42) 
[GCC 7.2.0] on linux2

I didn't use conda install to install caffe2. How can I confirm my root deriectoty of caffe2?(Is ~/caffe2/build? I am not sure.) Can anyone give me some ideas?
`
